core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -172,6 +173,11 @@ private void execute0(ClientRequestContext ctx, RedirectContext redirectCtx,
             return;
         }
 
+        if (initialAttempt) {
+            // If the original context had a selection timeout, propagate the exception for the first attempt.
```

Comment:
Add javadocs to `ClientRequestContext#newDerivedContext` which warns about this case.